src/user/routes/user.route.ts

```diff
@@ -0,0 +1,149 @@
+import { Request, Response, Router } from "express";
+import { Pool } from "pg";
+import HttpResponse from "../../common/models/response.model";
+import UserController from "../controllers/user.controller";
+import User from "../models/user.model";
+
+export default function UserRouter(): Router {
+  const userController: UserController = new UserController();
+  const router: Router = Router();
+
+  // Get All Users
+  // GET /user
+  // Optional Body = { limit: number } to limit the results.
+  router.get("/", (req: Request, res: Response) => {
+    const db: Pool = req.app.get("dbPool");
+    // TODO add functionality to handle LIMIT.
+    userController.getAllUsers(db).then(
+      (users: User[]) => {
+        const response: HttpResponse<User[]> = {
+          success: true,
+          returnCode: 200,
+          messages: [],
+          errors: [],
+          data: users,
+          rowCount: users.length,
+        };
+        res.status(200).send(response);
+      },
+      (err) => {
+        const response: HttpResponse = {
+          success: false,
+          returnCode: 500,
+          messages: [],
+          errors: [err],
```

Comment:
Can there be more than one error returned? Why have `errors` as an array instead of just `error`?

Reply:
True. My thought was `errors` would encapsulate all error messages, as some errors may have multiple lines (stack traces being a common multi line error message). We can change it to `errorMessages` for clarity.